src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.CSharp.cs

```diff
@@ -220,6 +220,59 @@ public static class C
                     modifiers: new Editing.DeclarationModifiers(isStatic: true));
             }
 
+            [Fact, Trait(Traits.Feature, Traits.Features.CodeGeneration), WorkItem(544405, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/544405")]
+            public async Task AddStaticAbstractClass()
+            {
+                var input = "namespace [|N|] { }";
+                var expected = @"namespace N
+{
+    public static class C
+    {
+    }
+}";
+                // note that 'abstract' is dropped here
+                await TestAddNamedTypeAsync(input, expected,
+                    modifiers: new Editing.DeclarationModifiers(isStatic: true, isAbstract: true));
+            }
+
+            [Theory, Trait(Traits.Feature, Traits.Features.CodeGeneration), WorkItem(544405, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/544405")]
+            [InlineData(Accessibility.NotApplicable)]
+            [InlineData(Accessibility.Internal)]
+            [InlineData(Accessibility.Public)]
+            public async Task AddFileClass(Accessibility accessibility)
+            {
+                var input = "namespace [|N|] { }";
+                var expected = @"namespace N
+{
+    file class C
+    {
+    }
+}";
+                // note: when invalid combinations of modifiers+accessibility are present here,
+                // we actually drop the accessibility. This is similar to what is done if someone declares a 'static abstract class C { }'.
+                await TestAddNamedTypeAsync(input, expected,
+                    accessibility: accessibility,
+                    modifiers: new Editing.DeclarationModifiers(isFile: true));
+            }
+
+            [Theory, Trait(Traits.Feature, Traits.Features.CodeGeneration), WorkItem(544405, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/544405")]
+            [InlineData("struct", TypeKind.Struct)]
+            [InlineData("interface", TypeKind.Interface)]
+            [InlineData("enum", TypeKind.Enum)]
+            public async Task AddFileType(string kindString, TypeKind typeKind)
```

Comment:
I wasn't able to figure out how to properly test 'delegate', 'record', or 'record struct' here.

- 'delegate' got me a test crash with a stacktrace consisting only of async thunks..
- I didn't see how to specify 'record' and 'record struct' in here.